openc3/spec/models/target_model_spec.rb

```diff
@@ -60,19 +66,132 @@ module OpenC3
       end
     end
 
+    describe "self.add_topics_to_microservice" do
+      xit "adds topics to microservices" do # "throws on missing topic, might be bug"
+        model = TargetModel.new(folder_name: "TEST", name: "INST", scope: "DEFAULT")
+        model.create
+        all = model.add_topics_to_microservice('INST', 'DEFAULT')
+        expect(all.keys).to contain_exactly("TEST", "SPEC")
+      end
+    end
+
     describe "self.all" do
       it "returns all the parsed targets" do
         model = TargetModel.new(folder_name: "TEST", name: "TEST", scope: "DEFAULT")
         model.create
         model = TargetModel.new(folder_name: "SPEC", name: "SPEC", scope: "DEFAULT")
         model.create
         all = TargetModel.all(scope: "DEFAULT")
+        expect(all).to_not be_nil
         expect(all.keys).to contain_exactly("TEST", "SPEC")
       end
     end
 
+    describe "render" do
+      it "renders" do
+        model = TargetModel.new(folder_name: "INST", name: "INST", scope: "DEFAULT")
+        model.create
+        model.render('_template.erb', {opt1: '1', opt2: '2', opt3: '3'})
+      rescue StandardError => e
```

Comment:
Expect an exception on a certain line, rather than rescue it.

Reply:
Global comment.

Reply:
I think I got them all.